src/test/java/in/projecteka/consentmanager/consent/ConsentRequestControllerTest.java

```diff
@@ -0,0 +1,96 @@
+package in.projecteka.consentmanager.consent;
+
+import in.projecteka.consentmanager.consent.model.response.ConsentRequestResponse;
+import in.projecteka.consentmanager.consent.repository.ConsentRequestRepository;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+import reactor.core.publisher.Mono;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@WebFluxTest(ConsentRequestController.class)
+@Import(ConsentRequestRepository.class)
+public class ConsentRequestControllerTest {
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @MockBean
+    private ConsentRequestRepository repository;
+
+
+
+    @Test
+    public void shouldAcceptConsentRequest() {
+
+        when(repository.insert(any(), any())).thenReturn(Mono.empty());
+
+        String body = "" +
```

Comment:
You can use the TestBuilder to build the request/response object.

Reply:
left it just like that for time being. There are some pros and cons using java constructs for payload. Will address later